src/webview/webViewEventHandlers.js

```diff
@@ -150,17 +153,28 @@ const handleImage = (props: Props, src: string, messageId: number) => {
   }
 };
 
-const handleLongPress = (props: Props, _: GetText, isHeader: boolean, messageId: number) => {
-  const message = props.messages.find(x => x.id === messageId);
-  if (!message) {
-    return;
+const handleLongPress = (
+  props: Props,
+  _: GetText,
+  target: string,
+  messageId: number,
+  href: string,
+) => {
+  if (target === 'link') {
```

Comment:
There's some redundancy here that creates a possibility of type errors, and with some adjustment of definitions we can make them impossible instead.

Specifically, once you see `target === 'link'` you're relying on the assumption that that means `href` will be valid. Which happens to be true in this version of the code, but it's not expressed in the types so the type-checker can't ensure it stays true, even though it's basically a type fact.

One way to fix this would be to just use `target: 'message'`, and let a non-`null` value of `href` mean it's actually a link. This would work OK.

Another would be to make another event type, like `copyLink`. This can have `href` required... and also it doesn't need `messageId`, which we aren't using here. This is a bit more work, but also more explicit.

Reply:
In the original version of this PR, I had implemented solution two, but @borisyankov suggested that 
> You might want to consider adding the href parameter to `MessageListEventLongPress` and add another target as `url`/`link`.

I agree with him, I don't think we require another event whose only difference with `longPress` is that it's a link, it'll increase code complexity unnecessarily.

Solution one suggests to use `target: 'message'` and check if `href` is not null, but I don't think this is the optimal approach, since it will make the code harder to understand (too abstract).
I have implemented the modified solution one (checking if href is not null) but I feel like I haven't understood the entire problem here. Could you please explain a bit more, or perhaps provide a link to an article which describes this in a more generic way?

Reply:
> I have implemented the modified solution one (checking if href is not null)

Hmm, I don't see an update -- did you forget to push? :wink: Or maybe I don't understand what you mean by this line.

Reply:
> Could you please explain a bit more, or perhaps provide a link to an article which describes this in a more generic way?

One common slogan for this principle is "make illegal states unrepresentable".

Here's one blog post I found (from a search just now) that explains it with another example:
https://oleb.net/blog/2018/03/making-illegal-states-unrepresentable/

Reply:
> Could you please explain a bit more, or perhaps provide a link to an article which describes this in a more generic way?

One common slogan for this principle is "make illegal states unrepresentable".

Here's one blog post I found (from a search just now) that explains it with another example:
https://oleb.net/blog/2018/03/making-illegal-states-unrepresentable/

Reply:
> Hmm, I don't see an update -- did you forget to push? 😉

Sorry about that, I meant I had those changes in the working directory.
Pushed now though!

Also thanks so much for the explanation! 😄